app/code/Magento/Backend/Model/Url.php

```diff
@@ -366,6 +366,18 @@ protected function _getMenu()
         return $this->_menu;
     }
 
+    /**
+     * Set scope entity
+     *
+     * @param null|int
+     * @return \Magento\Framework\UrlInterface
+     */
+    public function setScope($scopeId)
```

Comment:
Please call parent `setScope` as well to preserve existing functionality (and set `scope` DataObject key)

Also, as I can see \Magento\Framework\Url::setScope has `mixed $params` parameter